### PR TITLE
feat(mme): mme.yml configuration for amf

### PR DESCRIPTION
**Pull request by Lkishor123**
Signed-off-by: LKishor123 <laawanya.kishor@wavelabs.ai>

## Summary
Changes in generate_oai_config.py to enable configration of agw using "var/opt/magma/configs/mme.yml"

## Test Plan
UERANSIM:
 1. PDU session establishment:
 pdu_session

 2. mme_configration using: var/opt/magma/configs/mme.yml
mme_config

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 1]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     Returns: slice service type value
-    """
+    """  # noqa: DAR101, DAR201
+    enable_default_service_slice_type = get_service_config_value(
+        'mme', 'default_slice_service_type', None,
+    )
+
+    if enable_default_service_slice_type is not None:
+        return enable_default_service_slice_type
+
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: this is not needed as line 276 already takes care of returning a default value.

Comment (ulaskozat) [reply lte/gateway/python/scripts/generate_oai_config.py]: okay I guess you still want this to be configurable if it is defined in mme.yml file.

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 2]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     """
-    Retrieves default_slice_service_type config value,If it does not exist
+    Retrieve default_slice_service_type config value,If it does not exist
     it defaults to 1.
     Args:
         service_mconfig:
 
     Returns: slice service type value
-    """
+    """  # noqa: DAR101, DAR201
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: what is this `# noqa: DAR101, DAR201`?

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 3]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     """
-    Retrieves default_slice_differentiator config value,If it does not exist
+    Retrieve default_slice_differentiator config value,If it does not exist
     it defaults to 0xffffff.
     Args:
         service_mconfig:
 
     Returns: slice differentiator config value
-    """
+    """  # noqa: DAR101, DAR201
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: what is this `# noqa: DAR101, DAR201` ?

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 4]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     """
-    Retrieves amf_name config value,If it does not exist
+    Retrieve amf_name config value,If it does not exist
     it defaults to Default Name AMF_1.
     Args:
         service_mconfig:
 
     Returns: amf name string
-    """
+    """  # noqa: DAR101, DAR201
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: same comment.

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 5]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     """
-    Retrieves amf_region_id config value,If it does not exist
+    Retrieve amf_region_id config value,If it does not exist
     it defaults to Default value of 1
     Args:
         service_mconfig:
 
     Returns: amf region id
-    """
+    """  # noqa: DAR101, DAR201
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: same

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 6]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     """
-    Retrieves amf_set_id config value,If it does not exist
+    Retrieve amf_set_id config value,If it does not exist
     it defaults to Default String 1
     Args:
         service_mconfig:
 
     Returns: amf set id
-    """
+    """  # noqa: DAR101, DAR201
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: same comment

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 7]
```diff
@@ -259,73 +259,115 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
     """
-    Retrieves amf_pointer config value,If it does not exist
+    Retrieve amf_pointer config value,If it does not exist
     it defaults to Default String 1
     Args:
         service_mconfig:
 
     Returns: amf pointer value
-    """
+    """  # noqa: DAR101, DAR201
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: same comment

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 1 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: yes, it's required to fetch the configurations from mme.yml, hence code added.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 2 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: This stops, the doc string error which was coming for no reason.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 3 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: This stops, the doc string error which was coming for no reason.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 4 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: This stops, the doc string error which was coming for no reason.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 5 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: This stops, the doc string error which was coming for no reason.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 6 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: This stops, the doc string error which was coming for no reason.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 7 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: This stops, the doc string error which was coming for no reason.

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 8]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_default_service_slice_type
+
     return service_mconfig.default_slice_service_type or DEFAULT_NGAP_S_NSSAI_SST
 
 
 def _get_default_slice_differentiator_type_config(service_mconfig: object) -> str:
-    """
-    Retrieves default_slice_differentiator config value,If it does not exist
-    it defaults to 0xffffff.
+    """Retrieve default_slice_differentiator config value,If it does not exist it defaults to 0xffffff
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: can you replace `Retrieve default_slice_differentiator config value,If it does not exist it defaults to 0xffffff` with `Retrieve default_slice_differentiator config value. If it does not exist it defaults to 0xffffff.` Same punctuation and capitalization issues exist in other doc comments as well. Please fix them.

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 9]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_default_slice_differentiator_type
+
     return service_mconfig.default_slice_differentiator or DEFAULT_NGAP_S_NSSAI_SD
 
 
 def _get_amf_name_config(service_mconfig: object) -> str:
-    """
-    Retrieves amf_name config value,If it does not exist
-    it defaults to Default Name AMF_1.
+    """Retrieve amf_name config value,If it does not exist, it defaults to Default Name AMF_1
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: punctuation and capitalization issues (`,If` , missing `.` at the end of the sentence).

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 10]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_amf_name_config
+
     return service_mconfig.amf_name or DEFAULT_NGAP_AMF_NAME
 
 
 def _get_amf_region_id(service_mconfig: object) -> str:
-    """
-    Retrieves amf_region_id config value,If it does not exist
-    it defaults to Default value of 1
+    """Retrieve amf_region_id config value, If it does not exist it defaults to Default value of 1
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: punctuation and capitalization issues (`,If` , missing `.` at the end of the sentence).

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 11]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_amf_region_id
+
     return service_mconfig.amf_region_id or DEFAULT_NGAP_AMF_REGION_ID
 
 
 def _get_amf_set_id(service_mconfig: object) -> str:
-    """
-    Retrieves amf_set_id config value,If it does not exist
-    it defaults to Default String 1
+    """Retrieve amf_set_id config value, If it does not exist it defaults to Default String 1
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: punctuation and capitalization issues (`,If` , missing `.` at the end of the sentence).

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 12]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_amf_set_id
+
     return service_mconfig.amf_set_id or DEFAULT_NGAP_SET_ID
 
 
 def _get_amf_pointer(service_mconfig: object) -> str:
-    """
-    Retrieves amf_pointer config value,If it does not exist
-    it defaults to Default String 1
+    """Retrieve amf_pointer config value, If it does not exist it defaults to Default String 1
```
Comment (ulaskozat) [inline lte/gateway/python/scripts/generate_oai_config.py]: punctuation and capitalization issues (`,If` , missing `.` at the end of the sentence).

**Review by ulaskozat — CHANGES_REQUESTED** [review]

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 8 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 9 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 10 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 11 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 12 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 13]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
 
 
 def _get_default_slice_service_type_config(service_mconfig: object) -> str:
-    """
-    Retrieves default_slice_service_type config value,If it does not exist
-    it defaults to 1.
+    """Retrieve default_slice_service_type config value. If it does not exist, it defaults to 1.
```
Comment (ardzoht) [inline lte/gateway/python/scripts/generate_oai_config.py]: let's not use numerical values on function documentation
```suggestion
    """Retrieve default_slice_service_type config value. If it does not exist, it defaults to DEFAULT_NGAP_S_NSSAI_SST value.
```

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 14]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_default_service_slice_type
+
     return service_mconfig.default_slice_service_type or DEFAULT_NGAP_S_NSSAI_SST
 
 
 def _get_default_slice_differentiator_type_config(service_mconfig: object) -> str:
-    """
-    Retrieves default_slice_differentiator config value,If it does not exist
-    it defaults to 0xffffff.
+    """Retrieve default_slice_differentiator config value. If it does not exist it defaults to 0xffffff.
```
Comment (ardzoht) [inline lte/gateway/python/scripts/generate_oai_config.py]: same here, remove the value and use the variable name

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 15]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_amf_name_config
+
     return service_mconfig.amf_name or DEFAULT_NGAP_AMF_NAME
 
 
 def _get_amf_region_id(service_mconfig: object) -> str:
-    """
-    Retrieves amf_region_id config value,If it does not exist
-    it defaults to Default value of 1
+    """Retrieve amf_region_id config value. If it does not exist it defaults to Default value of 1.
```
Comment (ardzoht) [inline lte/gateway/python/scripts/generate_oai_config.py]: replace default numerical value on the function docs

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 16]
```diff
@@ -258,74 +258,116 @@ def _get_converged_core_config(service_mconfig: object) -> bool:
...
+        return enable_amf_region_id
+
     return service_mconfig.amf_region_id or DEFAULT_NGAP_AMF_REGION_ID
 
 
 def _get_amf_set_id(service_mconfig: object) -> str:
-    """
-    Retrieves amf_set_id config value,If it does not exist
-    it defaults to Default String 1
+    """Retrieve amf_set_id config value. If it does not exist it defaults to Default String 1.
```
Comment (ardzoht) [inline lte/gateway/python/scripts/generate_oai_config.py]: same here, update documentation

**Review by ardzoht — CHANGES_REQUESTED** [review]

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 13 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 14 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 15 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

File: lte/gateway/python/scripts/generate_oai_config.py (hunk 16 above)
Comment (Lkishor123) [reply lte/gateway/python/scripts/generate_oai_config.py]: Comment addressed.

**Review by ardzoht — APPROVED** [review]
lgtm

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
Looks like you forgot to sign and since you keep force pushing the single commit need to have the same semantic as the PR title.

**Review by ulaskozat — APPROVED** [review]
LGTM

**Pull request merged**